roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -976,13 +979,55 @@ RooAbsReal* RooAbsPdf::createNLL(RooAbsData& data, const RooLinkedList& cmdList)
   pc.defineDouble("IntegrateBins", "IntegrateBins", 0, -1.);
   pc.defineMutex("Range","RangeWithName") ;
   pc.defineMutex("GlobalObservables","GlobalObservablesTag") ;
+  pc.defineInt("NewStyle", "NewStyle", 0, 0);
+  pc.defineMutex(
+    "NewStyle",
+    "NumCPU"); // New style likelihoods define parallelization through Parallel(...) on RooMinimizer or fitTo.
+
 
   // Process and check varargs
   pc.process(cmdList) ;
   if (!pc.ok(true)) {
     return 0 ;
   }
 
+  if (pc.getInt("NewStyle")) {
+      int lut[3] = {2, 1, 0};
+      RooFit::TestStatistics::RooAbsL::Extended ext{
+        static_cast<RooFit::TestStatistics::RooAbsL::Extended>(lut[pc.getInt("ext")])};
+
+      RooArgSet cParsSet;
+      RooArgSet extConsSet;
+      RooArgSet glObsSet;
+
+      auto tmp = pc.getSet("cPars");
+      if (tmp)
+        cParsSet.add(*tmp);
+
+      tmp = pc.getSet("extCons");
+      if (tmp)
+        extConsSet.add(*tmp);
+
+      tmp = pc.getSet("glObs");
+      if (tmp)
+        glObsSet.add(*tmp);
+
+      const std::string rangeName = pc.getString("globstag", "", false);
+
+      RooFit::TestStatistics::ConstrainedParameters cPars(cParsSet);
+      RooFit::TestStatistics::ExternalConstraints extCons(extConsSet);
+      RooFit::TestStatistics::GlobalObservables glObs(glObsSet);
+
+      std::unique_ptr<RooFit::TestStatistics::RooAbsL> NewStyleL =
+        RooFit::TestStatistics::buildLikelihood(this, &data, ext, cPars, extCons, glObs, rangeName);
+
+      std::shared_ptr<RooFit::TestStatistics::RooAbsL> NewStyleL_shared = std::move(NewStyleL);
+      RooFit::TestStatistics::RooRealL *NewStyleL_real =
+        new RooFit::TestStatistics::RooRealL("likelihood", "", NewStyleL_shared);
+
+      return NewStyleL_real;
```

Comment:
```suggestion
      return new RooFit::TestStatistics::RooRealL("likelihood", "",
          RooFit::TestStatistics::buildLikelihood(this, &data, ext, cPars, extCons, glObs, rangeName));
```
I think you can do all of that smart pointer stuff implicitly, which would make the code less type-dependent and verbose